platform-sdk/docs/proposals/metric-labels/README.md

```diff
@@ -0,0 +1,325 @@
+# Metric labels
+
++author: @hendrikEbbers
++status: draft
+
+## Summary
+
+This proposal describes a new feature for the metrics module.
+The feature is called "metric labels" and allows to add labels to metrics.
+Labels are key-value pairs that can be used to filter and group metrics.
+Why this proposal uses the name "metric labels" (like prometheus does) other libraries (like micrometer or open telemetry) use the term tags or attributes.
+
+## Motivation
+
+Today the metrics module only supports the definition of metrics without adding custom labels.
+While each metric has a hardcoded category, name, and nodeId no other custom information can be added.
+This proposal suggests adding a new concept called "metric labels" to the metrics module and
+migrate the `category`, `name`, and `nodeId` information to labels.
+While the category and name are defined as String values the nodeId is a class called `NodeId`.
+By doing so the metrics module becomes more flexible and can be used 100% independent of the
+platform module since it does not depend on the platform module's `nodeId` anymore.
+
+## Goals
+
+- Add a new concept called "metric labels" to the metrics module
+- Migrate the `category`, `name`, and `nodeId` information to labels
+- Make the metrics module independent of the platform module
+- Provide a public API to define labels for metrics
+- Add labels to the snapshot API
+- The output of the snapshot API should contain the labels
+- The output of the snapshot API must be compatible with the current output (no breaking changes)
+- Create better and more dynamic dashboards in Grafana by using labels
+
+## Non-Goals
+
+Even with the new concept the metrics api should be only used in our (hashgraph) code.
+The metrics module should not be used by the end user.
+Here the preferred solution is still that we forward our metrics to a monitoring system that the custom application code uses.
+
+## Design & Architecture
+
+This chapter describes the changes that are needed to add labels to the metrics module.
+
+### About labels
+
+Before we start with the implementation we need to define what labels are.
+Labels are key-value pairs that can be used to filter and group metrics.
+Labels are used by monitoring systems like prometheus to create more dynamic dashboards.
+For example, a metric that counts the number of requests can have a label called `method` that defines if the request was a `GET` or `POST` request.
```

Comment:
so this is one metric or two? intuitively, I would say these are 2 metrics, one with a label `GET` and another with a label `POST`. this is conceptually confusing to me. can you add more detail explaining this